packages/nimble-components/src/table-column/duration-text/models/duration-formatter.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * A class for formatting a numeric value in the unit of seconds into a display value representing its
+ *  days, hours, minutes, and seconds.
+ */
+export class DurationFormatter {
+    private readonly daysFormatter: Intl.NumberFormat;
+    private readonly hoursFormatter: Intl.NumberFormat;
+    private readonly minutesFormatter: Intl.NumberFormat;
+    private readonly secondsFormatter: Intl.NumberFormat;
+    private readonly scientificSecondsFormatter: Intl.NumberFormat;
+
+    public constructor(private readonly lang: string) {
+        this.daysFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'day'
+        });
+        this.hoursFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'hour'
+        });
+        this.minutesFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'minute'
+        });
+        this.secondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second'
+        });
+        this.scientificSecondsFormatter = new Intl.NumberFormat(this.lang, {
+            style: 'unit',
+            unit: 'second',
+            notation: 'scientific',
+            maximumFractionDigits: 3
+        });
+    }
+
+    public format(value: number | null | undefined): string {
+        if (
+            value === null
+            || value === undefined
+            || value < 0
```

Comment:
I'm not sure. Ultimately, a difference between two timestamps can always be represented as a positive value, it just depends on which timestamp you subtract from which. Representing that difference as a negative value I don't think has any intrinsic value over representing it as the positive version of the difference, and semantically it just seems to make more sense as a positive value. A "duration" is a period of time _since_ a start time (which always moves forward). Like saying "the time since 0 seconds has been -2 seconds" is nonsensical.

I can certainly be convinced otherwise, but at the moment I don't think I feel compelled to support negative values.

Reply:
I don't have a concrete use case, but a user's data contains columns for timestamp `a`, timestamp `b`, and `b` - `a`, then you can end up with negative intervals if some `b`s preceed their `a`. Since I don't have a particular use case in mind, though, I'm satisfied with only supporting non-negative values now (and potentially adding support later if requested).